src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -50,38 +51,28 @@ private static void FillKnownMethodsGaps()
             }
         }
 
-        private static unsafe ulong GetAsciiStringAsLong(string str)
+        private static ulong GetAsciiStringAsLong(string str)
         {
             Debug.Assert(str.Length == 8, "String must be exactly 8 (ASCII) characters long.");
 
             var bytes = Encoding.ASCII.GetBytes(str);
 
-            fixed (byte* ptr = &bytes[0])
-            {
-                return *(ulong*)ptr;
-            }
+            return BinaryPrimitives.ReadUInt64LittleEndian(bytes);
```

Comment:
is it always going to be little endian?

Reply:
Yes https://github.com/dotnet/aspnetcore/blob/80861f1da944e1ef593bb5dfd36e882e50de32b0/src/Servers/Kestrel/Core/src/KestrelServer.cs#L129-L132

Reply:
If we ever do decide to try to support big endian architectures, we'll have a lot of code to fix up in Kestrel.